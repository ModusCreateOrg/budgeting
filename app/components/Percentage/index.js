import React from 'react';
import classNames from 'classnames';
import styles from './styles.scss';

const NUMBER_OF_DECIMALS = 2;

const Percentage = ({ balance, value }) => (
  <h2
    className={classNames({
      [styles.positive]: value >= 0,
      [styles.negative]: value < 0,
    })}
  >
    {value < 0 && '-'}
    {`${Number(value * 100 / balance).toFixed(NUMBER_OF_DECIMALS)}%`}
  </h2>
);

export default Percentage;
