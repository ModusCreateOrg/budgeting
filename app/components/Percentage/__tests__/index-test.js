import React from 'react';
import renderer from 'react-test-renderer';
import Percentage from '../';

it('renders correctly', () => {
  const tree = renderer.create(<Percentage balance={100} value={10} />).toJSON();
  expect(tree).toMatchSnapshot();
});
