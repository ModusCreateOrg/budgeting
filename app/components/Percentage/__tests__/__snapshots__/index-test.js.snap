// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h2
  className="positive"
>
  10.00%
</h2>
`;
